src/security/PeerConnector.cc

```diff
@@ -394,7 +397,14 @@ Security::PeerConnector::handleNegotiateError(const int ret)
         return;
 
     case SSL_ERROR_WANT_WRITE:
-        noteWantWrite();
+        if (needsValidationCallouts()) {
+            typedef NullaryMemFunT<Security::PeerConnector> cbDialer;
+            AsyncCall::Pointer resumeCall = JobCallback(83, 5,
+                                            cbDialer, this,
+                                            Security::PeerConnector::noteWantWrite);
+            suspendNegotiation(resumeCall);
```

Comment:
AFAICT, this check implies that we think that we have received the server certificate already and, hence, are now ready to fetch the missing certificates if needed. How do we know that `SSL_ERROR_WANT_WRITE` implies that we have received the server certificate already?

Reply:
Well again the code is not clear here.
This is works because the "needsValidationCallouts" will return true only if verifyData->missingIssuer is true. This is can be done only if the ssl_verify_cb is called, which is called only if certificates are received. 

I added a new method PeerConnector::certficatesReceived() which return true if we know that the certificates are received. This is the commit 6aff696457